packages/@blockprotocol/graph/src/react.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  RefObject,
+  useLayoutEffect,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+
+import {
+  BlockGraphProperties,
+  GraphBlockHandler,
+  GraphEmbedderHandler,
+} from "./index";
+
+export type BlockComponent<
+  Properties extends Record<string, unknown> | null = null,
+> = VoidFunctionComponent<BlockGraphProperties<Properties>>;
+
+export const useGraphBlockService = ({
+  callbacks,
+  ref,
+}: {
+  callbacks: ConstructorParameters<typeof GraphBlockHandler>[0]["callbacks"];
+  ref: RefObject<HTMLElement>;
+}) => {
+  const [graphService, setGraphService] = useState<GraphBlockHandler>();
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setGraphService(
+        new GraphBlockHandler({ callbacks, element: ref.current }),
+      );
+    }
+  }, [callbacks, ref]);
+
+  return { graphService };
```

Comment:
Why return this wrapped in an object? Why not just `return graphService` or `return [graphService]` if you want future extensibility? React hooks tend to leave it up to their caller to name fields. https://javascript.plainenglish.io/react-hooks-whats-the-difference-between-returning-an-array-and-an-object-34ccba62b71

Reply:
I don't think arrays are great for future extensibility because (a) less obvious what each entry is, (b) must retrieve unneeded variables if you don't want the 1st, 2nd etc.

The blog you link says:
> If our component will only have 1 (or few instances) of a hook, use object return.

What's the problem with having it an object? People can rename it if they want

Reply:
This is probably a stylistic preference, so will resolve this.